### PR TITLE
fixes the crash when duplicate book is added

### PR DESCRIPTION
**Pull request by 3bdifatah**
this change displays message when user tries to add duplicate book. @ewarlock if you suggestions for better message pls let me know. this is for issue #5.

### CODE REVIEW

**Comment by ewarlock** [comment]
This works as intended- the error message looks good as well! 
I think it is better to specify the error that is being raised. It looks like it is raising a BookError, which is a custom error class ? that needs to be imported from bookstore.py. This way DatabaseError and IntegrityError wouldn't need to be imported from sqlite3. 

So you could delete line 3 where you import DatabaseError and IntegrityError
And then add BookError to the list of things you import from bookstore:
`from bookstore import Book, BookStore, BookError`

Then on line 21 you specify `except BookError:` so you are specifying that this is what happens when we encounter BookError specifically.

I also want to ask @claraj is it okay to handle this in main()? Or would we want a different part of the program to handle it? I have been thinking about it but am not really sure where else to handle the error... maybe in add_book() ? It seems like it makes more sense to handle the error within the function that is adding the book but am not 100% sure.

**Comment by 3bdifatah** [comment]
thanks Ed, I actually didn't notice I imported DatabaseError and and integrityError until after I pushed the code. I was trying to figure out the error so I can specifically raise it but couldn't and it automatically import any error I tried. I will see what @Claraj says.

**Comment by ewarlock** [comment]
Got it what I have noticed in VSCode is that import statements are sometimes being automatically added to the top of the file. I have had to delete import statements that have appeared after modifying multiple files, it's really annoying. Maybe there is a fix for that in the editor somewhere ?

I ended up just letting the program crash and reading the crash message to figure out it was BookError.

**Comment by 3bdifatah** [comment]
@ewarlock the BookError you suggested is working now thanks. just let me know if there is anything I can improve?

**Comment by ewarlock** [comment]
I think it looks much better - I would say go ahead and merge unless Clara pops in with clarification where we would want to catch the error, since I'm not 100% sure on that myself.

File: main.py [hunk 1]
```diff
@@ -11,11 +11,14 @@ def main():
...
-        action()
-        if choice == 'Q':
-            break
+        try:
+            choice = ui.display_menu_get_choice(menu)
+            action = menu.get_action(choice)
+            action()
+            if choice == 'Q':
+                break
+        except BookError:
```
Comment (claraj) [inline main.py:20]: Sorry both, I completely missed the notification for this one. No, this is not where you would catch the BookError. Good practice to catch exceptions as close as possible to where they are created, so the try-except would be better in the function that adds the book.  What if another part of the program - like, deleting a book - and this raised a book error, and the program printed "The book you have entered is already in the database".  Would that be confusing?

**Review by claraj — COMMENTED** [review]
Apologies for missing the notification on this one.

**Pull request merged**